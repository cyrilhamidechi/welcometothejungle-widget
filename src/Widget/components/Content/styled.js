import styled from "styled-components";

import { colors } from "./../../../config";

export const Content = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
`;

export const Cover = styled(Content)`
  background-image: ${({ caption }) => caption && `url(${caption})`};
  background-size: cover;
  background-position: center;
  color: ${colors.white};
  transition: opacity 300ms ease;
`;

// Create wrapper with ratio 16/9
export const Wrapper = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  color: ${colors.white};
  overflow: hidden;

  &:hover {
    ${Cover} {
      opacity: 0.8;
    }
  }
`;

export const Button = styled.div`
  transform: scale(1);
  transition: transform 300ms ease;
`;

/* PICTURE COMPONENT */
export const Picture = styled.div`
  display: flex;
  opacity: 0;
  align-items: center;
  justify-content: center;
  height: 100%;

  &:hover {
    opacity: 1;

    ${Button} {
      transform: scale(1.5);
    }
  }
`;

/* VIDEO COMPONENT */
export const Video = styled.div`
  display: flex;
  justify-content: flex-end;
  flex-direction: column;
  height: 100%;

  &:hover {
    ${Button} {
      transform: scale(1.2);
    }
  }
`;

export const VideoItem = styled.div`
  display: flex;
  background-image: linear-gradient(transparent, rgba(0, 0, 0, 0.6));
  padding: 20px 10px 10px;
  font-size: 14px;
`;

export const VideoTitle = styled.div`
  margin-left: 10px;
`;

/* QUOTE COMPONENT */
export const QuoteIcon = styled.div`
  transition: margin-bottom 250ms ease;
`;

export const Quote = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  text-align: center;
  height: 100%;
  padding: 20px;
  box-sizing: border-box;
  font-size: 18px;

  &:hover {
    ${QuoteIcon} {
      margin-bottom: 5px;
    }
  }
`;
