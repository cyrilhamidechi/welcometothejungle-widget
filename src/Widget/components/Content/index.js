import PropTypes from "prop-types";
import React from "react";

import Button from "./../../../components/Button";
import Icon from "./../../../components/Icon";

import * as S from "./styled";

const Content = ({ item }) => (
  <S.Wrapper>
    <S.Cover caption={item.caption} />
    <S.Content>
      {item.type === "picture" && (
        <S.Picture>
          <S.Button>
            <Button iconName="picture" color="green" />
          </S.Button>
        </S.Picture>
      )}
      {item.type === "video" && (
        <S.Video>
          <S.VideoItem>
            <S.Button>
              <Button iconName="play" color="green" />
            </S.Button>
            <S.VideoTitle>
              <div>{item.title}</div>
              <div>{item.subtitle}</div>
            </S.VideoTitle>
          </S.VideoItem>
        </S.Video>
      )}
      {item.type === "quote" && (
        <S.Quote>
          <S.QuoteIcon>
            <Icon size="40" name="quote" color="green" />
          </S.QuoteIcon>
          {item.text}
        </S.Quote>
      )}
    </S.Content>
  </S.Wrapper>
);

Content.propTypes = {
  item: PropTypes.object.isRequired
};

export default Content;
