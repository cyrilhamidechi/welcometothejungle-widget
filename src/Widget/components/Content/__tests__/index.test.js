import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import Content from "../";

it("should render the component for picture", () => {
  const item = {
    type: "picture",
    caption: "img.png"
  };
  const tree = renderer.create(<Content item={item} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it("should render the component for video", () => {
  const item = {
    type: "video",
    caption: "img.png",
    title: "title",
    subtitle: "subtitle"
  };
  const tree = renderer.create(<Content item={item} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it("should render the component for quote", () => {
  const item = {
    type: "quote",
    text: "c'est pas faut !"
  };
  const tree = renderer.create(<Content item={item} />).toJSON();

  expect(tree).toMatchSnapshot();
});
