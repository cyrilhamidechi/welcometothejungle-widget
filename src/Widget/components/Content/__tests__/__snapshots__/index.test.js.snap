// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component for picture 1`] = `
.c8 {
  display: inline-block;
  line-height: 1;
  height: 20px;
}

.c8 svg {
  width: 100%;
  height: 100%;
}

.c8 path {
  -webkit-transition: fill 300ms;
  transition: fill 300ms;
}

.c8 svg {
  width: 20px;
  height: 20px;
}

.c8 svg path {
  fill: #fff;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 30px;
  height: 30px;
  padding: 0;
  border: 0;
  border-radius: 30px;
  background: #00c29a;
  color: #fff;
  font-size: 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-transition: background 300ms ease-in-out,opacity 300ms;
  transition: background 300ms ease-in-out,opacity 300ms;
}

.c7:hover:enabled {
  background: #37cbaa;
}

.c7:disabled {
  opacity: 0.3;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: "10px";
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-image: url(img.png);
  background-size: cover;
  background-position: center;
  color: #fff;
  -webkit-transition: opacity 300ms ease;
  transition: opacity 300ms ease;
}

.c0 {
  position: relative;
  width: 100%;
  height: 100%;
  color: #fff;
  overflow: hidden;
}

.c0:hover .c1 {
  opacity: 0.8;
}

.c6 {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: -webkit-transform 300ms ease;
  -webkit-transition: transform 300ms ease;
  transition: transform 300ms ease;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
}

.c4:hover {
  opacity: 1;
}

.c4:hover .c5 {
  -webkit-transform: scale(1.5);
  -ms-transform: scale(1.5);
  transform: scale(1.5);
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  />
  <div
    className="c3"
  >
    <div
      className="c4"
    >
      <div
        className="c5 c6"
      >
        <button
          className="c7"
          color="green"
          onClick={[Function]}
        >
          <span
            className="c8"
            color="white"
            size="20"
          >
            <svg
              aria-labelledby="title"
              height="1em"
              role="img"
              title="picture"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M18 15v4c0 .55-.45 1-1 1H5c-.55 0-1-.45-1-1V7c0-.55.45-1 1-1h3.02c.55 0 1-.45 1-1s-.45-1-1-1H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-5c0-.55-.45-1-1-1s-1 .45-1 1zm-2.5 3H6.52c-.42 0-.65-.48-.39-.81l1.74-2.23a.5.5 0 0 1 .78-.01l1.56 1.88 2.35-3.02c.2-.26.6-.26.79.01l2.55 3.39c.25.32.01.79-.4.79zm3.8-9.11c.48-.77.75-1.67.69-2.66-.13-2.15-1.84-3.97-3.97-4.2A4.5 4.5 0 0 0 11 6.5c0 2.49 2.01 4.5 4.49 4.5.88 0 1.7-.26 2.39-.7l2.41 2.41c.39.39 1.03.39 1.42 0 .39-.39.39-1.03 0-1.42l-2.41-2.4zM15.5 9a2.5 2.5 0 0 1 0-5 2.5 2.5 0 0 1 0 5z"
              />
            </svg>
          </span>
          <span
            className="c9"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render the component for quote 1`] = `
.c7 {
  display: inline-block;
  line-height: 1;
  height: 40px;
}

.c7 svg {
  width: 100%;
  height: 100%;
}

.c7 path {
  -webkit-transition: fill 300ms;
  transition: fill 300ms;
}

.c7 svg {
  width: 40px;
  height: 40px;
}

.c7 svg path {
  fill: #00c29a;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-size: cover;
  background-position: center;
  color: #fff;
  -webkit-transition: opacity 300ms ease;
  transition: opacity 300ms ease;
}

.c0 {
  position: relative;
  width: 100%;
  height: 100%;
  color: #fff;
  overflow: hidden;
}

.c0:hover .c1 {
  opacity: 0.8;
}

.c6 {
  -webkit-transition: margin-bottom 250ms ease;
  transition: margin-bottom 250ms ease;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  height: 100%;
  padding: 20px;
  box-sizing: border-box;
  font-size: 18px;
}

.c4:hover .c5 {
  margin-bottom: 5px;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  />
  <div
    className="c3"
  >
    <div
      className="c4"
    >
      <div
        className="c5 c6"
      >
        <span
          className="c7"
          color="green"
          size="40"
        >
          <svg
            aria-labelledby="title"
            height="1em"
            role="img"
            title="quote"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M7.17 17c.51 0 .98-.29 1.2-.74l1.42-2.84c.14-.28.21-.58.21-.89V8c0-.55-.45-1-1-1H5c-.55 0-1 .45-1 1v4c0 .55.45 1 1 1h2l-1.03 2.06c-.45.89.2 1.94 1.2 1.94zm10 0c.51 0 .98-.29 1.2-.74l1.42-2.84c.14-.28.21-.58.21-.89V8c0-.55-.45-1-1-1h-4c-.55 0-1 .45-1 1v4c0 .55.45 1 1 1h2l-1.03 2.06c-.45.89.2 1.94 1.2 1.94z"
            />
          </svg>
        </span>
      </div>
      c'est pas faut !
    </div>
  </div>
</div>
`;

exports[`should render the component for video 1`] = `
.c9 {
  display: inline-block;
  line-height: 1;
  height: 20px;
}

.c9 svg {
  width: 100%;
  height: 100%;
}

.c9 path {
  -webkit-transition: fill 300ms;
  transition: fill 300ms;
}

.c9 svg {
  width: 20px;
  height: 20px;
}

.c9 svg path {
  fill: #fff;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 30px;
  height: 30px;
  padding: 0;
  border: 0;
  border-radius: 30px;
  background: #00c29a;
  color: #fff;
  font-size: 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-transition: background 300ms ease-in-out,opacity 300ms;
  transition: background 300ms ease-in-out,opacity 300ms;
}

.c8:hover:enabled {
  background: #37cbaa;
}

.c8:disabled {
  opacity: 0.3;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: "10px";
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-image: url(img.png);
  background-size: cover;
  background-position: center;
  color: #fff;
  -webkit-transition: opacity 300ms ease;
  transition: opacity 300ms ease;
}

.c0 {
  position: relative;
  width: 100%;
  height: 100%;
  color: #fff;
  overflow: hidden;
}

.c0:hover .c1 {
  opacity: 0.8;
}

.c7 {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: -webkit-transform 300ms ease;
  -webkit-transition: transform 300ms ease;
  transition: transform 300ms ease;
}

.c12:hover .c6 {
  -webkit-transform: scale(1.5);
  -ms-transform: scale(1.5);
  transform: scale(1.5);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

.c4:hover .c6 {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-image: linear-gradient(transparent,rgba(0,0,0,0.6));
  padding: 20px 10px 10px;
  font-size: 14px;
}

.c11 {
  margin-left: 10px;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  />
  <div
    className="c3"
  >
    <div
      className="c4"
    >
      <div
        className="c5"
      >
        <div
          className="c6 c7"
        >
          <button
            className="c8"
            color="green"
            onClick={[Function]}
          >
            <span
              className="c9"
              color="white"
              size="20"
            >
              <svg
                aria-labelledby="title"
                height="1em"
                role="img"
                title="play"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  d="M8 6.82v10.36c0 .79.87 1.27 1.54.84l8.14-5.18a1 1 0 0 0 0-1.69L9.54 5.98A.998.998 0 0 0 8 6.82z"
                />
              </svg>
            </span>
            <span
              className="c10"
            />
          </button>
        </div>
        <div
          className="c11"
        >
          <div>
            title
          </div>
          <div>
            subtitle
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
